src/tokens.js

```diff
@@ -0,0 +1,279 @@
+// @flow
+
+import type { List } from 'immutable';
+
+import type Syntax from './syntax';
+
+import { TokenType, TokenClass } from 'shift-parser/dist/tokenizer';
```

Comment:
is it possible to remove this dependency and use our own `TokenType` and `TokenClass` objects?